File: app/src/main/java/maderski/markparkingspot/MainActivity.java
package maderski.markparkingspot;

import android.Manifest;
import android.app.Activity;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.support.v4.app.ActivityCompat;

public class MainActivity extends Activity {
    private static final String TAG = MainActivity.class.getName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if(hasLocationPermission()) {
            Actions actions = new Actions(this);
            actions.checkIfCanGetLocation();
            finish();
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        finish();
    }

    private boolean hasLocationPermission() {
        PackageManager packageManager = getPackageManager();
        int hasPermission = packageManager.checkPermission(Manifest.permission.ACCESS_FINE_LOCATION,
                getPackageName());
        //Check if Permission is granted
        if(hasPermission != PackageManager.PERMISSION_GRANTED){
            ActivityCompat.requestPermissions(
                    this,
                    new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, PackageManager.PERMISSION_GRANTED);
            return false;
        }else{
            return true;
        }
    }
}

File: app/src/main/java/maderski/markparkingspot/MainActivity.kt
package maderski.markparkingspot

import android.app.Activity
import android.os.Bundle

class MainActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val isLocationPermissionGranted = PermissionUtils.hasLocationPermission(this)
        if (isLocationPermissionGranted) {
            val actions = Actions(this)
            actions.checkIfCanGetLocation()
            finish()
        }
    }

    override fun onResume() {
        super.onResume()
        finish()
    }
}
